File: src/test/java/com/uapp_llc/test/comparator/ColumnComparator.java
package com.uapp_llc.test.comparator;

import java.util.Comparator;

import com.google.common.collect.ComparisonChain;

import com.uapp_llc.model.Column;

class ColumnComparator implements Comparator<Column> {

  @Override
  public int compare(Column left, Column right) {
    return ComparisonChain.start()
        .compare(left.getId(), right.getId())
        .compare(left.getName(), right.getName())
        .compare(left.getIndex(), right.getIndex())
        .result();
  }

}

File: src/test/kotlin/com/uapp_llc/test/comparator/ColumnComparator.kt
package com.uapp_llc.test.comparator

import com.google.common.collect.ComparisonChain
import com.uapp_llc.model.Column
import java.util.Comparator

class ColumnComparator : Comparator<Column> {

  override fun compare(left: Column?, right: Column?): Int {
    return ComparisonChain.start()
        .compare(left?.id as Comparable<*>, right?.id as Comparable<*>)
        .compare(left.name as Comparable<*>, right.name as Comparable<*>)
        .compare(left.index as Comparable<*>, right.index as Comparable<*>)
        .result()
  }

}
